File: 数组/二分查找/version1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const search = function (nums, target) {
    let left = 0;
    let right = nums.length - 1;
    while (left <= right) {
        const middle = Math.floor((left + right) / 2);
        if (nums[middle] < target) {
            left = middle + 1; // target 在右区间，所以 [middle + 1, right]
        } else if (nums[middle] > target) {
            right = middle - 1; // target 在左区间，所以 [left, middle - 1]
        } else {
            return middle;
        }
    }
    return -1;
};

File: 数组/二分查找/version1.go
package main

func search(nums []int, target int) int {
    left := 0
    right := len(nums) - 1
    for left <= right {
        middle := left + (right - left) / 2
        if nums[middle] == target {
            return middle
        } else if nums[middle] > target {
            right = middle - 1
        } else {
            left = middle + 1
        }
    }
    return -1
}

func main() {
    b :=[5]int{1,2,3,4,5}
    a := search(b, 2)
}
